src/github.com/stellar/gateway/bridge/handlers/request_handler_payment_test.go

```diff
@@ -788,39 +769,147 @@ func TestRequestHandlerPayment(t *testing.T) {
 					assert.Equal(t, values.Encode(), params.Encode())
 				}).Once()
 
-				statusCode, response := getResponse(testServer, params)
+				statusCode, response := net.GetResponse(testServer, params)
 				responseString := strings.TrimSpace(string(response))
 				assert.Equal(t, 500, statusCode)
-				expectedResponse := horizon.SubmitTransactionResponse{Error: horizon.ServerError}
-				assert.Equal(t, expectedResponse.Marshal(), []byte(responseString))
+				assert.Equal(t, "{\n  \"code\": \"internal_server_error\",\n  \"message\": \"Internal Server Error, please try again.\"\n}", responseString)
+			})
+
+			Convey("it should return denied when compliance server returns denied", func() {
+				mockHttpClient.On(
+					"PostForm",
+					"http://compliance/send",
+					mock.AnythingOfType("url.Values"),
+				).Return(
+					net.BuildHttpResponse(200, "{\"auth_response\": {\"tx_status\": \"denied\"}}"),
+					nil,
+				).Run(func(args mock.Arguments) {
+					values := args.Get(1).(url.Values)
+					// bridge server does not send source seed to compliance
+					assert.Equal(t, []string{"GAW77Z6GPWXSODJOMF5L5BMX6VMYGEJRKUNBC2CZ725JTQZORK74HQQD"}, values["source"])
+					values.Del("source")
+					params.Del("source")
+					assert.Equal(t, values.Encode(), params.Encode())
+				}).Once()
+
+				statusCode, response := net.GetResponse(testServer, params)
+				responseString := strings.TrimSpace(string(response))
+				assert.Equal(t, 403, statusCode)
+				assert.Equal(t, "{\n  \"code\": \"denied\",\n  \"message\": \"Transaction denied by destination.\"\n}", responseString)
 			})
 
-			// Convey("it should submit transaction when compliance server returns success", func() {
```

Comment:
Could you leave a note alongside commented out blocks of code like this?  Personally I prefer to leave the reason why it's commented out and the conditions that need to be satisfied for it to be re-enabled.